config/v2/config.go

```diff
@@ -4,54 +4,72 @@ import (
 	"encoding/json"
 	"fmt"
 	"math/rand"
+	"path/filepath"
 	"reflect"
 
 	v1 "github.com/chanzuckerberg/fogg/config/v1"
 	"github.com/chanzuckerberg/fogg/errs"
+	"github.com/spf13/afero"
+	"gopkg.in/yaml.v2"
 )
 
-func ReadConfig(b []byte) (*Config, error) {
+//ReadConfig take a byte array as input and outputs a json or yaml config file
+func ReadConfig(fs afero.Fs, b []byte, configFile string) (*Config, error) {
+	var e error
 	c := &Config{
 		Docker: false,
 	}
-	e := json.Unmarshal(b, c)
 
-	return c, errs.WrapUser(e, "unable to parse config")
+	info, e := fs.Stat(configFile)
+	if e != nil {
+		return nil, errs.WrapUser(e, "unable to find file")
+	}
+
+	//Determines the extension of the file
+	switch filepath.Ext(info.Name()) {
+	case ".yml", ".yaml":
+		e = yaml.Unmarshal(b, c)
+	case ".json":
+		e = json.Unmarshal(b, c)
+	default:
+		return nil, errs.NewUser("File type is not supported")
```

Comment:
```suggestion
return nil, errs.NewUserf("File type %s is not supported", filepath.Ext(info.Name()))
```

... but refactor so that you don't have to run filepath.Ext(info.Name()) twice